crates/biome_js_analyze/src/aria_analyzers/nursery/no_static_element_interactions.rs

```diff
@@ -0,0 +1,126 @@
+use std::{collections::HashMap, hash::BuildHasherDefault};
+
+use crate::aria_services::Aria;
+use biome_analyze::{context::RuleContext, declare_rule, Rule, RuleDiagnostic};
+use biome_aria::AriaRoles;
+use biome_console::markup;
+use biome_js_syntax::jsx_ext::AnyJsxElement;
+use biome_rowan::AstNode;
+use rustc_hash::FxHasher;
+
+declare_rule! {
+    /// Static elements should not be interactive.
+    ///
+    /// Static HTML elements do not have semantic meaning. This is clear in the case of `<div>` and `<span>`. It is less so clear in the case of elements that seem semantic, but that do not have a semantic mapping in the accessibility layer. For example `<a>`, `<big>`, `<blockquote>`, `<footer>`, `<picture>`, `<strike>` and `<time>` -- to name a few -- have no semantic layer mapping. They are as void of meaning as `<div>`.
+    ///
+    /// The [WAI-ARIA role attribute](https://www.w3.org/TR/wai-aria-1.1/#usage_intro) confers a semantic mapping to an element. The semantic value can then be expressed to a user via assistive technology.
+    /// In order to add interactivity such as a mouse or key event listener to a static element, that element must be given a role value as well.
+    ///
+    /// Source: https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/main/docs/rules/no-static-element-interactions.md#jsx-a11yno-static-element-interactions
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```jsx,expect_diagnostic
+    /// <div onClick={() => {}} />;
+    /// ```
+    ///
+    /// ## Valid
+    ///
+    /// ```jsx
+    /// <div role="button" onClick={() => {}}  />;
+    /// ```
+    ///
+    /// ```jsx
+    /// <button onClick={() => {}} className="foo" />;
+    /// ```
+    ///
+    /// ```jsx
+    /// <input type="text" onClick={() => {}} />;
+    /// ```
+    ///
+    pub(crate) NoStaticElementInteractions {
+        version: "next",
+        name: "noStaticElementInteractions",
+        recommended: false,
+    }
+}
+
+const INTERACTIVE_HANDLER_LIST: [&str; 6] = [
+    "onClick",
+    "onMouseDown",
+    "onMouseUp",
+    "onKeyDown",
+    "onKeyPress",
+    "onKeyUp",
+];
```

Comment:
Do we want to match parity with the [eslint implementation](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/main/docs/rules/no-static-element-interactions.md#rule-options) and add an option to the rule for consumer to modify this list as  or leave it hardcoded?

Reply:
Personally, I think the introduction of the option is counterproductive because it can create a pretext to introduce more errors. If the user has a specific exception, they should suppress the rule via comments.

What do you think?

Reply:
I think I would agree with that.